tests/unit/dataSource/properties/tests.html

```diff
@@ -536,22 +628,113 @@
 		});
 
 		test("Test getDetachedRecord API method", function () {
+			var arrayData = [
+				[0, "Bread", 2.5],
+				[1, "Milk", 3.5],
+				[2, "Vint soda", 20.9]
+			];
+			dsLocal.settings.localSchemaTransform = false;
+			dsLocal.settings.dataSource = arrayData;
 			dsLocal.settings.primaryKey = null;
-			var t = dsLocal.setCellValue(0, "Name", "NewVal", true);
-			var rec = dsLocal.getDetachedRecord(t);
 
-			equal(rec.Name, "NewVal", "Detached value should be updated.");
+			dsLocal.dataBind();
+			stop();
+
+			setTimeout(function () {
+				start();
+				var t = dsLocal.setCellValue(0, "Name", "NewVal", true);
+				var rec = dsLocal.getDetachedRecord(t);
+
+				equal(rec.Name, "NewVal", "Detached value should be updated.");
```

Comment:
You should probably call setCellValue once again on the same record and then check that the detached record is not updated.